src/egon/data/datasets/power_plants/pv_rooftop_buildings.py

```diff
@@ -0,0 +1,2433 @@
+    ).to_crs(to_crs)
+
+
+def synthetic_buildings(
+    to_crs: CRS,
+) -> gpd.GeoDataFrame:
+    """
+    Read synthetic buildings data from eGo^n Database.
+    Parameters
+    -----------
+    to_crs : pyproj.crs.crs.CRS
+        CRS to transform geometries to.
+    Returns
+    -------
+    geopandas.GeoDataFrame
+        GeoDataFrame containing OSM buildings data.
+    """
+    with db.session_scope() as session:
+        query = session.query(
+            OsmBuildingsSynthetic.id,
+            OsmBuildingsSynthetic.area,
+            OsmBuildingsSynthetic.geom_point.label("geom"),
+        )
+
+    return gpd.read_postgis(
+        query.statement, query.session.bind, index_col="id"
+    ).to_crs(to_crs)
+
+
+def add_ags_to_buildings(
+    buildings_gdf: gpd.GeoDataFrame,
+    municipalities_gdf: gpd.GeoDataFrame,
+) -> gpd.GeoDataFrame:
+    """
+    Add information about AGS ID to buildings.
+    Parameters
+    -----------
+    buildings_gdf : geopandas.GeoDataFrame
+        GeoDataFrame containing OSM buildings data.
+    municipalities_gdf : geopandas.GeoDataFrame
+        GeoDataFrame with municipality data.
+    Returns
+    -------
+    gepandas.GeoDataFrame
+        GeoDataFrame containing OSM buildings data
+        with AGS ID added.
+    """
+    return buildings_gdf.sjoin(
+        municipalities_gdf,
+        how="left",
+        predicate="intersects",
+    ).rename(columns={"index_right": "ags"})
+
+
+def drop_buildings_outside_muns(
+    buildings_gdf: gpd.GeoDataFrame,
+) -> gpd.GeoDataFrame:
+    """
+    Drop all buildings outside of municipalities.
+    Parameters
+    -----------
+    buildings_gdf : geopandas.GeoDataFrame
+        GeoDataFrame containing OSM buildings data.
+    Returns
+    -------
+    gepandas.GeoDataFrame
+        GeoDataFrame containing OSM buildings data
+        with buildings without an AGS ID dropped.
+    """
+    gdf = buildings_gdf.loc[~buildings_gdf.ags.isna()]
+
+    logger.debug(
+        f"{len(buildings_gdf) - len(gdf)} "
+        f"({(len(buildings_gdf) - len(gdf)) / len(buildings_gdf) * 100:g}%) "
+        f"of {len(buildings_gdf)} values are outside of the municipalities "
+        "and are therefore dropped."
+    )
+
+    return gdf
+
+
+def load_building_data():
+    """Read buildings from DB
+    Tables:
+    * `openstreetmap.osm_buildings_filtered` (from OSM)
+    * `openstreetmap.osm_buildings_synthetic` (synthetic, creaed by us)
+    Use column `id` for both as it is unique hence you concat both datasets.
+    """
+
+    municipalities_gdf = municipality_data()
+
+    osm_buildings_gdf = osm_buildings(municipalities_gdf.crs)
+
+    synthetic_buildings_gdf = synthetic_buildings(municipalities_gdf.crs)
+
+    buildings_gdf = gpd.GeoDataFrame(
+        pd.concat(
+            [
+                osm_buildings_gdf,
+                synthetic_buildings_gdf,
+            ]
+        ),
+        crs=osm_buildings_gdf.crs,
+    ).rename(columns={"area": "building_area"})
```

Comment:
On second thought, I'm not sure whether it makes much sense to allocate the plants to synthetic buildings too as they have an artificial/arbitrary ground area of 25 sqm. It might be good to discuss this again...